lib/publiccloud/azure.pm

```diff
@@ -388,14 +393,15 @@ sub create_image_definition {
     my $gen = ($sku =~ "gen2" ? "V2" : "V1");
     my $tags = generate_tags();
 
-    my $arch = (get_var('PUBLIC_CLOUD_ARCH', get_required_var('ARCH'))) eq 'x86_64' ? 'x64' : 'Arm64';
+    my $image = compose_azure_image_name();
     my $publisher = get_var("PUBLIC_CLOUD_AZURE_PUBLISHER", "qe-c");
-    my $offer = get_var("PUBLIC_CLOUD_AZURE_OFFER", get_var('DISTRI') . '-' . get_var('VERSION') . '-' . get_var('FLAVOR') . '-' . $arch);
+    my $offer = get_var("PUBLIC_CLOUD_AZURE_OFFER", $image);
 
     my $definition = $self->get_image_definition($resource_group, $gallery);
     if (defined $definition) {
         record_info("use img-def", "Using found image definitions:\n$definition");
     } else {
+        my $arch = (split("-", $image))[-1];
```

Comment:
Here the `arch` is retrieved by the first back in `$distri-$version-$flavor-$arch`

Reply:
As agreed, in order to allow indipendent image name handling and avoid future definition-changes could impact this value, I will change this functionality to retrive arch value.